Shape_detection/include/CGAL/Shape_detection/Region_growing/Segment_set/Line_segment_region.h

```diff
@@ -0,0 +1,300 @@
+      (max_angle * static_cast<FT>(CGAL_PI)) / FT(180))));
+    const FT cos_value = parameters::choose_parameter(
+      parameters::get_parameter(np, internal_np::cosine_of_maximum_angle), default_cos_value);
+    CGAL_precondition(cos_value >= FT(0) && cos_value <= FT(1));
+    m_cos_value_threshold = cos_value;
+    m_squared_cos_value_threshold = m_cos_value_threshold * m_cos_value_threshold;
+  }
+
+  /// @}
+  /// \name Access
+  /// @{
+
+  /*!
+    \brief implements `RegionType::region_index_map()`.
+
+    This function creates an empty property map that maps iterators on the input range `Item` to `std::size_t`.
+  */
+  Region_index_map region_index_map() {
+    return Region_index_map(m_region_map);
+  }
+
+  /*!
+    \brief implements `RegionType::primitive()`.
+
+    This function provides the support line of the seed segment.
+
+    \return Primitive parameters that fits the region
+
+    \pre is_valid_region(region)
+    \pre update(region)
+  */
+  Primitive primitive() const {
+    // Compute the primitive (line) from the seed segment on the fly
+    const auto& seed_seg = get(m_segment_map, m_seed);
+    const Point& seed_s = seed_seg.source();
+    const Point& seed_t = seed_seg.target();
+    return Line(seed_s, seed_t);
+  }
+
+  /*!
+    \brief implements `RegionType::is_part_of_region()`.
+
+    This function determines if a segment with the index `query` is within
+    the `maximum_distance` from the corresponding line and if the angle between the
+    direction of this segment and the line's direction is within the `maximum_angle`.
+    If both conditions are satisfied, it returns `true`, otherwise `false`.
+
+    \param query item of the query segment
+  */
+  bool is_part_of_region(const Item query,
+                         const Region&) const {
+    const auto& seg = get(m_segment_map, query);
+    const Point& s = seg.source();
+    const Point& t = seg.target();
+    const Vector dir(s, t);
+    if (m_distance_threshold > FT(0)) {
+      const auto& seed_seg = get(m_segment_map, m_seed);
+      const Point& seed_s = seed_seg.source();
+      const Point& seed_t = seed_seg.target();
+      Line seed_line(seed_s, seed_t);
+      if (m_compare_squared_distance(s, seed_line, m_squared_distance_threshold) != SMALLER ||
+          m_compare_squared_distance(t, seed_line, m_squared_distance_threshold) != SMALLER)
+        return false;
+    }
+    if (m_cos_value_threshold < FT(1)) {
+      const auto& seed_seg = get(m_segment_map, m_seed);
+      const Point& seed_s = seed_seg.source();
+      const Point& seed_t = seed_seg.target();
+      Vector seed_dir(seed_s, seed_t);
+      FT cos_value = m_scalar_product(dir, seed_dir);
+      FT squared_cos_value = cos_value * cos_value;
+      FT threshold = m_squared_cos_value_threshold;
+      threshold *= m_squared_length(dir);
+      threshold *= m_squared_length(seed_dir);
+      if (squared_cos_value < threshold)
```

Comment:
this is not a predicate if we don't have exact constructions. We need something like in the case of faces.

Reply:
should be correct now